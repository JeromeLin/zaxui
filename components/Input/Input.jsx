import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import Autosize from 'autosize';

class Input extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      length: (props.value || props.defaultValue || '').length,
    };
    this.onInputChange = this.onInputChange.bind(this);
  }

  componentDidMount() {
    this.initAutosize();
  }

  componentDidUpdate(prevProps) {
    if (prevProps.style !== this.props.style ||
        prevProps.className !== this.props.className) {
      Autosize.update(this.input);
    }
  }

  componentWillUnmount() {
    this.destroyAutosize();
  }

  onInputChange(e) {
    const { onChange } = this.props;

    this.setState({
      length: e.target.value.length,
    });

    typeof onChange === 'function' && onChange(e);
  }

  // 初始化自适应高度
  initAutosize() {
    const { autosize } = this.props;
    autosize && Autosize(this.input);
  }

  // 销毁自适应高度
  destroyAutosize() {
    const { autosize } = this.props;
    autosize && Autosize.destroy(this.input);
  }

  render() {
    const {
      prefixCls,
      className,
      placeholder,
      type,
      maxLength,
      disabled,
      autosize,
      showLength,
      ...others
    } = this.props;

    const cls = classnames({
      [`${prefixCls}`]: true,
      [className]: !!className,
      disabled,
    });

    const inputRender = (type === 'textarea')
      ? (
        <textarea
          {...others}
          ref={(ele) => { this.input = ele; }}
          className={cls}
          placeholder={placeholder}
          disabled={disabled}
          maxLength={maxLength}
          onChange={this.onInputChange}
          />
      )
      : (
        <input
          {...others}
          ref={(ele) => { this.input = ele; }}
          type={type}
          className={cls}
          placeholder={placeholder}
          disabled={disabled}
          maxLength={maxLength}
          onChange={this.onInputChange}
          />
      );

    const valueRender = (type === 'date')
      ? <div className={`${prefixCls}-placeholder`}>{placeholder}</div>
      : null;

    const textLengthRender = (showLength && maxLength)
      ? <div className={`${prefixCls}-length`}>{`${this.state.length}/${maxLength}`}</div>
      : null;

    return (
      <div className={cls}>
        {valueRender}
        {inputRender}
        {textLengthRender}
      </div>
    );
  }
}

Input.propTypes = {
  prefixCls: PropTypes.string,
  className: PropTypes.string,
  type: PropTypes.string,
  disabled: PropTypes.bool,
  rows: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  autosize: PropTypes.bool,
  showLength: PropTypes.bool,
  onChange: PropTypes.func,
};

Input.defaultProps = {
  prefixCls: 'za-input',
  className: null,
  type: 'text',
  disabled: false,
  rows: null,
  autosize: false,
  showLength: false,
  onChange() {},
};

export default Input;
