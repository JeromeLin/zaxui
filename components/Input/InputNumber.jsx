import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import Events from '../utils/events';
import Keyboard from '../Keyboard';

class InputNumber extends Component {

  constructor(props) {
    super(props);
    this.state = {
      visible: false,
      value: props.value || '',
    };
  }

  componentDidMount() {
    Events.on(document.body, 'touchstart', this.onClosePicker);
  }

  componentWillUnmount() {
    Events.off(document.body, 'touchstart', this.onClosePicker);
  }

  onClosePicker = (e) => {
    if (!this.picker) return;
    if (!this.state.visible) return;

    const pNode = ((node) => {
      while (node.parentNode && node.parentNode !== document.body) {
        if (node === this.picker) {
          return node;
        }
        node = node.parentNode;
      }
    })(e.target);

    !pNode && this.close();
  }

  onFocus = () => {
    document.activeElement.blur();
    this.setState({ visible: true });
  }

  onKeyClick = (key) => {
    const value = this.state.value;
    const newValue = (key === 'delete')
      ? value.slice(0, value.length - 1)
      : value + key;

    this.setState({ value: newValue });
  }

  open = () => {
    this.setState({ visible: true });
  }

  close = () => {
    this.setState({ visible: false });
  }

  render() {
    const { prefixCls, className, type, disabled, ...others } = this.props;
    const { visible, value } = this.state;

    const cls = classnames(prefixCls, `${prefixCls}-number`, className, disabled);

    return (
      <div className={cls} {...others} ref={(ele) => { this.picker = ele; }}>
        <input
          {...others}
          ref={(ele) => { this.input = ele; }}
          type="text"
          value={value}
          disabled={disabled}
          onFocus={this.onFocus}
          />
        <Keyboard.Picker
          type={type}
          visible={visible}
          onKeyClick={this.onKeyClick}
          />
      </div>
    );
  }
}

InputNumber.propTypes = {
  prefixCls: PropTypes.string,
  className: PropTypes.string,
  value: PropTypes.string,
  defaultValue: PropTypes.string,
  disabled: PropTypes.bool,
  maxLength: PropTypes.number,
  onChange: PropTypes.func,
};

InputNumber.defaultProps = {
  prefixCls: 'za-input',
  disabled: false,
};

export default InputNumber;
