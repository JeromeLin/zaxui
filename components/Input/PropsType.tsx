export interface BaseInputProps {
  type?: string;
  autosize?: boolean;
}

export interface BaseInputTextProps {
  value?: string;
  defaultValue?: string;
  disabled?: boolean;
  maxLength?: number;
  onChange?: (x?: any) => void;
}

export interface BaseInputNumberProps {
  type?: 'number' | 'price' | 'idcard';
  value?: string | number;
  defaultValue?: string | number;
  disabled?: boolean;
  maxLength?: number;
  onChange?: (x?: any) => void;
}

export interface BaseInputTextareaProps {
  type?: 'number' | 'price' | 'idcard';
  value?: string;
  defaultValue?: string;
  disabled?: boolean;
  rows?: number;
  autosize?: boolean;
  maxLength?: number;
  showLength?: boolean;
  style?: React.CSSProperties;
  onChange?: (x?: any) => void;
}
