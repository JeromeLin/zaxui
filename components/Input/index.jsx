import React, { Component, PropTypes } from 'react';
import classnames from 'classnames';
import { Icon } from '../../components';

// 清除输入内容
class Input extends Component {
  constructor(props) {
    super(props);

    const { placeholder, type, isRadius, isDisabled, size, defaultValue, className, ...others } = props;

    const disabled = 'disabled' in props || isDisabled;
    const radius = 'radius' in props || isRadius;

    this.state = {
      value: defaultValue,
      showClearBtn: false,
      counter: 0,
    };

        // 绑定函数的this
    ['handleBlur', 'handleFocus', 'handleChange', 'clearInputText'].forEach((elem) => {
      this[elem] = this[elem].bind(this);
    });

    const cls = this.cls = classnames({
      'ui-input': true,
      disabled,
      radius,
      [`size-${size}`]: size,
      [className]: !!className,
    });

        // valueText?
    this.valueText = (type === 'date') ? <div className="ui-input-placeholder">{placeholder}</div> : null;

        // 输入框
    this.input = function getInput() {
      return (type === 'textarea')
                ? <textarea
                  {...others}
                  className={cls}
                  placeholder={placeholder}
                  disabled={disabled}>{defaultValue}</textarea>
                : <input
                  {...others}
                  type={type}
                  className={cls}
                  placeholder={placeholder}
                  defaultValue={defaultValue}
                  value={this.state.value}
                  disabled={disabled}
                  onFocus={this.handleFocus}
                  onBlur={this.handleBlur}
                  onChange={this.handleChange}
                  />;
    };

        // 只有text || textarea 类型下才出现Icon
        // 使用onMouseDown处理 click点击之前就失去焦点的问题
    this.clearBtn = type === 'text' ?
            (<Icon
              className="clear-input-btn"
              type="wrong-round-fill"
              onMouseDown={this.clearInputText}
              onTouchStart={this.clearInputText}
              theme="error"
              />) : null;
  }


    // 获取焦点
  handleFocus(e) {
    if (this.state.value !== '') {
      this.setState({
        showClearBtn: true,
      });
    }

        // 处理外部props的事件
    if (this.props.onFocus) {
      this.props.onFocus.apply(this, arguments);
    }
  }

    // 失去焦点
  handleBlur(e) {
    this.setState({
      showClearBtn: false,
    });

  // 处理外部props的事件
    typeof this.props.onBlur === 'function' && this.props.onBlur.apply(this, arguments);
  }


    // 处理输入事件
  handleChange(e) {
    const { value } = e.currentTarget;

    const stateTemp = {
      value,
    };
    if (value === '') {
      stateTemp.showClearBtn = false;
    } else if (this.state.showClearBtn !== true) {
      stateTemp.showClearBtn = true;
    }

    this.setState(stateTemp);

        // 处理外部事件
    typeof this.props.onChange === 'function' && this.props.onChange.apply(this, arguments);
  }

    // 清除输入内容
  clearInputText(e) {
    e.preventDefault();
    this.setState({
      value: '',
      showClearBtn: false,
    });
  }

  render() {
    return (
      <span
        className={this.cls} style={{
          padding: '5px 0',
        }}>
        {this.valueText}
        {this.input()}
        {this.state.showClearBtn ? this.clearBtn : null}
      </span>
    );
  }

}

Input.propTypes = {
  type: PropTypes.string,
  size: PropTypes.oneOf(['xl', 'lg', 'sm', 'xs']),
  isRadius: PropTypes.bool,
  isDisabled: PropTypes.bool,
  className: PropTypes.string,
};

Input.defaultProps = {
  type: 'text',
  size: null,
  isRadius: false,
  isDisabled: false,
  className: null,
  defaultValue: '',
};

export default Input;
