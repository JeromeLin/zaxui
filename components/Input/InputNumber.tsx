import React, { Component } from 'react';
import classnames from 'classnames';
import { BaseInputNumberProps } from './PropsType';
import Events from '../utils/events';
import KeyboardPicker from '../KeyboardPicker';

declare const document;

export interface InputNumberProps extends BaseInputNumberProps {
  prefixCls?: string;
  className?: string;
}

export default class InputNumber extends Component<InputNumberProps, any> {

  static defaultProps = {
    prefixCls: 'za-input',
    disabled: false,
  };

  private input;

  constructor(props) {
    super(props);
    this.state = {
      visible: false,
      value: props.value || props.defaultValue || '',
    };
  }

  componentDidMount() {
    Events.on(document.body, 'touchstart', this.onClosePicker);
  }

  componentWillUnmount() {
    Events.off(document.body, 'touchstart', this.onClosePicker);
  }

  closest = (el, selector) => {
    const matchesSelector = el.matches || el.webkitMatchesSelector || el.mozMatchesSelector || el.msMatchesSelector;
    while (el) {
      if (matchesSelector.call(el, selector)) {
        return el;
      } else {
        el = el.parentElement;
      }
    }
    return null;
  }

  onClosePicker = (e) => {
    if (!this.input || !this.state.visible) {
      return;
    }

    const pNode = this.closest(e.target, '.za-keyboard');

    if (!pNode) {
      this.close();
    }
  }

  onFocus = () => {
    document.activeElement.blur();
    this.open();
  }

  onKeyClick = (key) => {
    const value = this.state.value;
    const newValue = (key === 'delete')
      ? value.slice(0, value.length - 1)
      : value + key;

    if (newValue !== value) {
      const { onChange } = this.props;
      this.setState({ value: newValue });

      if (typeof onChange === 'function') {
        onChange(newValue);
      }
    }
  }

  open = () => {
    this.setState({ visible: true });
  }

  close = () => {
    this.setState({ visible: false });
  }

  render() {
    const { prefixCls, className, type, disabled, defaultValue, placeholder, ...others } = this.props;
    const { visible, value } = this.state;

    const cls = classnames(prefixCls, `${prefixCls}-number`, className, {
      disabled,
      focus: visible,
    });

    return (
      <div className={cls} ref={(ele) => { this.input = ele; }} onClick={this.onFocus}>
        {!value && <div className={`${prefixCls}-placeholder`}>{placeholder}</div>}
        <div className={`${prefixCls}-content`}>{value}</div>
        <input
          {...others}
          type="hidden"
          value={value}
          disabled={disabled}
          onFocus={this.onFocus}
        />
        <KeyboardPicker
          visible={visible}
          type={type}
          onKeyClick={this.onKeyClick}
        />
      </div>
    );
  }
}
