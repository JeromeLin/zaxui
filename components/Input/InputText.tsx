import React, { PureComponent } from 'react';
import classnames from 'classnames';
import { BaseInputTextProps } from './PropsType';

export interface InputTextProps extends BaseInputTextProps {
  prefixCls?: string;
  className?: string;
}

export default class InputText extends PureComponent<InputTextProps, {}> {

  static defaultProps = {
    prefixCls: 'za-input',
    disabled: false,
  };

  onInputChange = (e) => {
    const { onChange } = this.props;
    if (typeof onChange === 'function') {
      onChange(e.target.value);
    }
  }

  render() {
    const { prefixCls, className, disabled, ...others } = this.props;
    const cls = classnames(prefixCls, `${prefixCls}-text`, className, {
      disabled,
    });

    return (
      <div className={cls}>
        <input
          {...others}
          type="text"
          disabled={disabled}
          onChange={this.onInputChange}
        />
      </div>
    );
  }
}
