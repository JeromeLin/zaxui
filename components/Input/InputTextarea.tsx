import React, { PureComponent } from 'react';
import classnames from 'classnames';
import Autosize from 'autosize';
import { BaseInputTextareaProps } from './PropsType';

const regexAstralSymbols = /[\uD800-\uDBFF][\uDC00-\uDFFF]|\n/g;

function countSymbols(text = '') {
  return text.replace(regexAstralSymbols, '_').length;
}

export interface InputTextareaProps extends BaseInputTextareaProps {
  prefixCls?: string;
  className?: string;
}

export default class InputTextarea extends PureComponent<InputTextareaProps, any> {

  static defaultProps = {
    prefixCls: 'za-input',
    disabled: false,
    autosize: false,
    showLength: false,
  };

  private input;

  constructor(props) {
    super(props);
    this.state = {
      length: (props.value || props.defaultValue || '').length,
    };
  }

  componentDidMount() {
    this.initAutosize();
  }

  componentDidUpdate(prevProps) {
    this.updateAutosize(prevProps);
  }

  componentWillUnmount() {
    this.destroyAutosize();
  }

  onInputChange = (e) => {
    const { onChange } = this.props;
    const value = e.target.value;
    const length = countSymbols(value) + (value ? value.length : 0);
    this.setState({ length });
    if (typeof onChange === 'function') {
      onChange(e);
    }
  }

  // 初始化自适应高度
  initAutosize = () => {
    const { autosize } = this.props;
    if (autosize) {
      Autosize(this.input);
    }
  }

  updateAutosize = (prevProps) => {
    if (prevProps.style !== this.props.style || prevProps.className !== this.props.className) {
      Autosize.update(this.input);
    }
  }

  // 销毁自适应高度
  destroyAutosize = () => {
    const { autosize } = this.props;
    if (autosize) {
      Autosize.destroy(this.input);
    }
  }

  render() {
    const {
      prefixCls,
      className,
      maxLength,
      disabled,
      autosize,
      showLength,
      ...others,
    } = this.props;

    const cls = classnames(prefixCls, `${prefixCls}-textarea`, className, {
      disabled,
    });

    const textLengthRender =
      showLength &&
      maxLength &&
      <div className={`${prefixCls}-length`}>
        {`${this.state.length}/${maxLength}`}
      </div>;

    return (
      <div className={cls}>
        <textarea
          {...others}
          ref={(ele) => { this.input = ele; }}
          disabled={disabled}
          maxLength={maxLength}
          onChange={this.onInputChange}
        />
        {textLengthRender}
      </div>
    );
  }
}
