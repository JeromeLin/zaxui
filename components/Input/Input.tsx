import React, { PureComponent } from 'react';
import { BaseInputProps } from './PropsType';
import InputNumber from './InputNumber';
import InputBase from './InputBase';
import InputTextarea from './InputTextarea';

export interface InputProps extends BaseInputProps {
}

export default class Input extends PureComponent<BaseInputProps, {}> {

  static defaultProps = {
    type: 'text',
  };

  render() {
    const { type, ...others } = this.props;
    switch (type) {
      case 'idcard':
      case 'price':
      case 'number':
        return <InputNumber {...others} type={type} />;

      case 'textarea':
        return <InputTextarea {...others} />;

      default:
        return <InputBase {...others} type={type} />;
    }
  }

}
