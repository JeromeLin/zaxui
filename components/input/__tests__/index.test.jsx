import React from 'react';
import { findDOMNode } from 'react-dom';
import { render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Input from '../index';
import InputBase from '../InputBase';


describe('Input', () => {
  it('renders correctly', () => {
    const wrapper = render(<Input type="text" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('showLength', () => {
    const wrapper = render(<Input showLength maxLength={100} type="textarea" rows={4} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders onFocus called correctly', () => {
    const onFocus = jest.fn();
    const wrapper = mount(
      <Input onFocus={onFocus} />
    );
    wrapper.find('input[type="text"]').simulate('focus');
    expect(onFocus).toBeCalled();
    // expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });

  it('renders onBlur called correctly', () => {
    const wrapper = mount(<InputBase />);
    const spy = jest.spyOn(wrapper.instance(), 'onBlur');
    wrapper.find('input').simulate('focus');
    wrapper.find('input').simulate('blur');
    expect(spy).toHaveBeenCalled();
  });

  it('renders onClear called correctly', () => {
    const onClear = jest.fn();
    const wrapper = mount(
      <Input
        value=""
        onClear={onClear}
        clearable
      />
    );

    const input = wrapper.find('input[type="text"]');
    input.simulate('change', { target: { value: 'My new value' } });
    wrapper.find('i.za-input-clear').simulate('click');
    expect(onClear).toHaveBeenCalled();
    expect(input.instance().value).toEqual('');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  // it('renders cn', () => {
  //   const onFocus = jest.fn();
  //   const wrapper = mount(
  //     <Input
  //       value=""
  //       onFocus={onFocus}
  //     />
  //   );

  //   const input = wrapper.find('input[type="text"]');
  //   input.simulate('change', { target: { value: '测试' } });
  //   wrapper.find('i.za-input-clear').simulate('click');
  //   expect(input.instance().value).toEqual('');
  //   expect(onFocus).toHaveBeenCalled();
  //   expect(toJson(wrapper)).toMatchSnapshot();
  // });
});

describe('Input.Textarea', () => {
  it('renders correctly', () => {
    const wrapper = render(<Input type="textarea" rows={4} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('autoHeight', () => {
    jest.useFakeTimers();
    const props = {
      autoHeight: true,
      type: 'textarea',
      rows: 4,
      value: 'foo',
      onChange: jest.fn(),
    };
    const wrapper = mount(<Input {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.find('textarea').simulate('change', { target: { value: 'this is a test!' } });
    expect(props.onChange).toBeCalledWith('this is a test!');
    jest.runAllTimers();
    wrapper.unmount();
  });

  it('renders onFocus called correctly', () => {
    const onFocus = jest.fn();
    const wrapper = mount(
      <Input type="textarea" onFocus={onFocus} />
    );
    wrapper.find('textarea').simulate('focus');
    expect(onFocus).toBeCalled();
    // expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });

  it('renders onBlur called correctly', () => {
    const onBlur = jest.fn();
    const wrapper = mount(<Input type="textarea" onBlur={onBlur} />);
    // const spy = jest.spyOn(wrapper.instance(), 'onBlur');
    wrapper.find('textarea').simulate('focus');
    wrapper.find('textarea').simulate('blur');
    expect(onBlur).toHaveBeenCalled();
  });
});

describe('Input.Number', () => {
  it('renders correctly', () => {
    const wrapper = mount(<Input type="number" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders onFocus called correctly', () => {
    const onFocus = jest.fn();
    const wrapper = mount(
      <Input type="number" onFocus={onFocus} />
    );
    wrapper.find('.za-input-content').simulate('click');
    expect(onFocus).toBeCalled();
    // expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });

  it('renders onClear called correctly', () => {
    const onClear = jest.fn();
    const wrapper = mount(
      <Input
        type="number"
        onClear={onClear}
      />
    );

    const input = wrapper.find('input[type="hidden"]');
    input.simulate('change', { target: { value: 'My new value' } });
    wrapper.find('i.za-input-clear').simulate('click');
    expect(onClear).toHaveBeenCalled();
    expect(input.instance().value).toEqual('');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('enter number', () => {
    const onChange = jest.fn();
    const wrapper = mount(<Input type="number" focused onChange={onChange} />);
    wrapper.find('input').simulate('focus');
    const keys = wrapper.find('.za-keyboard-keys');
    keys.childAt(0).simulate('click');
    expect(onChange).toBeCalledWith('1');
    wrapper.unmount();
  });

  it('input number hidden', () => {
    const onBlur = jest.fn();
    const wrapper = mount(<Input type="number" focused onBlur={onBlur} />);
    wrapper.find('input').simulate('focus');
    const keys = wrapper.find('.za-keyboard-keys');
    keys.childAt(11).simulate('click');
    expect(onBlur).toBeCalled();
    wrapper.unmount();
  });

  it('input number hidden', () => {
    const map = {};
    document.body.addEventListener = jest.fn((event, cb) => {
      map[event] = cb;
    });
    const onBlur = jest.fn();
    const wrapper = mount(<Input type="number" focused onBlur={onBlur} />);
    map.click({
      target: findDOMNode(document.body), // eslint-disable-line
    });
    expect(onBlur).toBeCalled();
    wrapper.unmount();
  });
});

describe('Input.Price', () => {
  it('renders correctly', () => {
    const wrapper = mount(<Input type="price" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('Input.Idcard', () => {
  it('renders correctly', () => {
    const wrapper = mount(<Input type="idcard" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
