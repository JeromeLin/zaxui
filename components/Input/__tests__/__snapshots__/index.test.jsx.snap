// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<div
  class="za-input za-input-text"
>
  <input
    type="text"
  />
</div>
`;

exports[`Input showLength 1`] = `
<div
  class="za-input za-input-textarea"
>
  <textarea
    maxlength="100"
    rows="4"
  />
  <div
    class="za-input-length"
  >
    0/100
  </div>
</div>
`;

exports[`Input.Idcard renders correctly 1`] = `
<div
  class="za-input za-input-number"
>
  <input
    type="text"
    value=""
  />
  <div
    class="za-popup za-popup-hidden"
  >
    <div
      class="za-popup-wrapper za-popup-wrapper-bottom"
      style="-webkit-transition-duration:200ms;transition-duration:200ms"
    >
      <div
        class="za-keyboard"
      >
        <div
          class="za-keyboard-key"
        >
          <div
            class="za-keyboard-key-item"
          >
            1
          </div>
          <div
            class="za-keyboard-key-item"
          >
            2
          </div>
          <div
            class="za-keyboard-key-item"
          >
            3
          </div>
          <div
            class="za-keyboard-key-item"
          >
            4
          </div>
          <div
            class="za-keyboard-key-item"
          >
            5
          </div>
          <div
            class="za-keyboard-key-item"
          >
            6
          </div>
          <div
            class="za-keyboard-key-item"
          >
            7
          </div>
          <div
            class="za-keyboard-key-item"
          >
            8
          </div>
          <div
            class="za-keyboard-key-item"
          >
            9
          </div>
          <div
            class="za-keyboard-key-item"
          >
            x
          </div>
          <div
            class="za-keyboard-key-item"
          >
            0
          </div>
          <div
            class="za-keyboard-key-item"
          >
            <i
              class="za-icon za-icon-keyboard"
            />
          </div>
        </div>
        <div
          class="za-keyboard-handle"
        >
          <div
            class="za-keyboard-handle-item"
          >
            <i
              class="za-icon za-icon-deletekey"
            />
          </div>
          <div
            class="za-keyboard-handle-item za-keyboard-handle-item-ok"
          >
            确定
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Input.Number renders correctly 1`] = `
<div
  class="za-input za-input-number"
>
  <input
    type="text"
    value=""
  />
  <div
    class="za-popup za-popup-hidden"
  >
    <div
      class="za-popup-wrapper za-popup-wrapper-bottom"
      style="-webkit-transition-duration:200ms;transition-duration:200ms"
    >
      <div
        class="za-keyboard"
      >
        <div
          class="za-keyboard-key"
        >
          <div
            class="za-keyboard-key-item"
          >
            1
          </div>
          <div
            class="za-keyboard-key-item"
          >
            2
          </div>
          <div
            class="za-keyboard-key-item"
          >
            3
          </div>
          <div
            class="za-keyboard-key-item"
          >
            4
          </div>
          <div
            class="za-keyboard-key-item"
          >
            5
          </div>
          <div
            class="za-keyboard-key-item"
          >
            6
          </div>
          <div
            class="za-keyboard-key-item"
          >
            7
          </div>
          <div
            class="za-keyboard-key-item"
          >
            8
          </div>
          <div
            class="za-keyboard-key-item"
          >
            9
          </div>
          <div
            class="za-keyboard-key-item za-keyboard-key-item-disabled"
          />
          <div
            class="za-keyboard-key-item"
          >
            0
          </div>
          <div
            class="za-keyboard-key-item"
          >
            <i
              class="za-icon za-icon-keyboard"
            />
          </div>
        </div>
        <div
          class="za-keyboard-handle"
        >
          <div
            class="za-keyboard-handle-item"
          >
            <i
              class="za-icon za-icon-deletekey"
            />
          </div>
          <div
            class="za-keyboard-handle-item za-keyboard-handle-item-ok"
          >
            确定
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Input.Price renders correctly 1`] = `
<div
  class="za-input za-input-number"
>
  <input
    type="text"
    value=""
  />
  <div
    class="za-popup za-popup-hidden"
  >
    <div
      class="za-popup-wrapper za-popup-wrapper-bottom"
      style="-webkit-transition-duration:200ms;transition-duration:200ms"
    >
      <div
        class="za-keyboard"
      >
        <div
          class="za-keyboard-key"
        >
          <div
            class="za-keyboard-key-item"
          >
            1
          </div>
          <div
            class="za-keyboard-key-item"
          >
            2
          </div>
          <div
            class="za-keyboard-key-item"
          >
            3
          </div>
          <div
            class="za-keyboard-key-item"
          >
            4
          </div>
          <div
            class="za-keyboard-key-item"
          >
            5
          </div>
          <div
            class="za-keyboard-key-item"
          >
            6
          </div>
          <div
            class="za-keyboard-key-item"
          >
            7
          </div>
          <div
            class="za-keyboard-key-item"
          >
            8
          </div>
          <div
            class="za-keyboard-key-item"
          >
            9
          </div>
          <div
            class="za-keyboard-key-item"
          >
            .
          </div>
          <div
            class="za-keyboard-key-item"
          >
            0
          </div>
          <div
            class="za-keyboard-key-item"
          >
            <i
              class="za-icon za-icon-keyboard"
            />
          </div>
        </div>
        <div
          class="za-keyboard-handle"
        >
          <div
            class="za-keyboard-handle-item"
          >
            <i
              class="za-icon za-icon-deletekey"
            />
          </div>
          <div
            class="za-keyboard-handle-item za-keyboard-handle-item-ok"
          >
            确定
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Input.Textarea autosize 1`] = `
<Input
  autosize={true}
  onChange={[Function]}
  rows={4}
  type="textarea"
  value="foo"
>
  <InputTextarea
    autosize={true}
    disabled={false}
    onChange={[Function]}
    prefixCls="za-input"
    rows={4}
    showLength={false}
    value="foo"
  >
    <div
      className="za-input za-input-textarea"
    >
      <textarea
        disabled={false}
        maxLength={undefined}
        onChange={[Function]}
        rows={4}
        value="foo"
      />
    </div>
  </InputTextarea>
</Input>
`;

exports[`Input.Textarea renders correctly 1`] = `
<div
  class="za-input za-input-textarea"
>
  <textarea
    rows="4"
  />
</div>
`;
